[cleowenxuan] iP

cleowenxuan opened the pull request:
# Asher 🤖 
> "Either you run the day or the day runs you." - Jim Rohn 
[source](https://www.shopify.com/blog/motivational-quotes)

Allow Asher to help you remember the things you need to do! It is,
* text-based
* easy to learn
* ~~FAST~~ SUPER FAST to use

All you need to do is, 
1. download it from [here](https://nus-cs2103-ay2324s2.github.io/website/schedule/week4/project.html)
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you. 

And it is **FREE**!

Features:

- [x] Manage your todos! 📚  
- [x] Manage deadlines  🖥️ 
- [x] Manage events 📆 
- [ ] Reminders ⏳ (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here is the `main` method:

```
    public static void main(String[] args) {
        String dataFile = "./taskLists.txt";
        new Asher(dataFile).run();
    }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

Commit:
Level0 Completed

Commit:
Level1 Completed

Commit:
Level2 Completed

Commit:
Level3 Completed

Commit:
Level4 Completed

Commit:
Automated Text UI Testing Completed

Commit:
Level 5 Completed

Commit:
Level 6 Completed

Commit:
Final Touches

Commit:
Level 7 Completed

Commit:
Merge branch 'branch-Level-7'

Commit:
Level 8 Completed

Commit:
Merge branch 'branch-Level-8'
Merge Level8 with master.

ruijietay commented on src/main/java/Asher.java:
```diff
@@ -0,0 +1,326 @@
+        try {
+            File file = new File(filePath);
+            if (!file.exists()) {
+                boolean created = file.createNewFile();
```
Should this be renamed with a prefix preceding the boolean variable `created`?

ruijietay commented on src/main/java/Asher.java:
```diff
@@ -0,0 +1,326 @@
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Scanner;
+import java.util.List;
+import java.util.ArrayList;
+import java.time.LocalDate;
+import java.time.LocalTime;
```
Would it be better to have the import statements be in alphabetical order?

ruijietay commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,37 @@
+    @Override
+    public String writeToString() {
+        String status = isDone ? "1" : "0";
+        return "E" + " | " + status + " | " + description + " | " + startDate + " | " + startTime + " | " + endDate + " | " + endTime;
```
Would this be better if it was split into two lines?

ruijietay commented on src/main/java/Asher.java:
```diff
@@ -0,0 +1,326 @@
+        if (taskNumber != -1) {
+            tasks.get(taskNumber).markDone();
+            System.out.println("Nice! I've marked this task as done:");
+            System.out.println(" "  + tasks.get(taskNumber));
```
Should the extra space between " " and + be removed?

ruijietay reviewed (commented):
Generally, great job following the coding standards! However, I feel that it would be better to have whitespaces between certain characters such as operators and braces. I believe it would help in ensuring your code follows a coding standard by installing the checkstyle plugin on IntellJ! 😄

Commit:
A-MoreOOP Completed

Commit:
Merge branch 'A-MoreOOP'
Merge branch A-MoreOOP with the master.

Commit:
A-Packages Completed

Commit:
Merge branch 'add-gradle-support'
Merging add-gradle-support with master.

Commit:
A-Gradle Completed

Commit:
A-JUnit Completed

Commit:
A-Jar Completed

Commit:
A-JavaDoc Completed

Commit:
Merge branch 'branch-A-JavaDoc'
Merge Java Doc branch with the master.

Commit:
A-CodingStandard Completed

Commit:
Resolve Conflicts between CodingStandard branch and JavaDoc branch

Commit:
Level 9 Completed

Commit:
Resolve Conflicts with Level-9 and master

PateShin commented on src/main/java/asher/Commands/Storage.java:
```diff
@@ -0,0 +1,125 @@
+     * @param tasks The List of tasks.
+     */
+
+    public void getFileContents(String filePath, TaskList tasks) {
```
Consider to remove the ```String filePath``` as class variable is used, not a parameter.

PateShin commented on src/main/java/asher/Asher.java:
```diff
@@ -0,0 +1,60 @@
+     */
+    public void run() {
+        String dataFile = "./taskLists.txt";
+        storage.getFileContents(dataFile, tasks);
```
Perhaps consider using```getFileTasks``` as **tasks** are scanned from ```.txt``` file.

PateShin commented on src/main/java/asher/Tasks/Deadline.java:
```diff
@@ -0,0 +1,56 @@
+    }
+
+    @Override
+    public String writeToString() {
```
Consider renaming ```toFileStringFormat``` to be more descriptive.

PateShin commented on src/main/java/asher/Ui/Ui.java:
```diff
@@ -0,0 +1,104 @@
+     * Displays the number of tasks in the list message.
+     * @param totalTasks The total number of tasks.
+     */
+    public void showNumberOfTaskInListMessage(int totalTasks) {
```
I like how you described the function with its name

PateShin commented on src/main/java/asher/Tasks/Event.java:
```diff
@@ -0,0 +1,80 @@
+
+    @Override
+    public String toString() {
+        DateTimeFormatter formattingDate = DateTimeFormatter.ofPattern("MMM dd yyyy");
+        DateTimeFormatter formattingTime = DateTimeFormatter.ofPattern("hh:mm a");
```
I like how you named them. Perhaps try ```DateFormatted``` and ```FormattedTime```?

PateShin reviewed (commented):
Codes are mostly conforms to the Code Quality: Naming standards Great job 👍 
There are just a few minor issues and suggestions that could be improved.

Commit:
JavaFX Tutorial Guide

Commit:
Level-10 Completed, has some minor bugs

Commit:
Resolve issues with merging branch-Level-10

Commit:
A-Assertions Completed

Commit:
A-CodeQuality Completed

Some codes do not follow the coding standard.

Refactor the packages in the asher package to follow the package naming convention.
Refactor the java files inside test.java.asher to follow the class naming convention.
Rephrase method/class header comments to follow the format specified in the coding standard.
Java Docs are being added to most methods and classes for clarity.

cleowenxuan committed:
Merge pull request #2 from cleowenxuan/branch-A-Assertions

Merge the branch branch-A-Assertions with master.

Commit:
Merge branch 'master' into branch-A-CodeQuality.

cleowenxuan committed:
Merge pull request #3 from cleowenxuan/branch-A-CodeQuality

Merge the branch branch-A-CodeQuality with master.

Commit:
BCD-Extensions Completed

Commit:
Merge branch 'branch-BCD-Extension'

Commit:
User Guide Completed

Commit:
Fix get file content issue

Commit:
Fix issues regarding DialogBox.java

Commit:
Modify the extension into a command instead of sorting at TaskList

Commit:
A-AiAssisted Completed

Commit:
Modify User Guide

Commit:
Add Ui.png

Commit:
Fix bugs for DialogBox

Commit:
Fix minor bugs

Commit:
Fix exception bugs
